evap/evaluation/tests/test_views.py

```diff
@@ -246,3 +248,43 @@ def test_notebook(self):
 
         user.refresh_from_db()
         self.assertEqual(user.notes, self.note)
+
+
+class TestResetEvaluation(WebTestStaffMode):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.manager = make_manager()
+        cls.semester = baker.make(Semester, results_are_archived=True)
+
+    def reset_from_x_to_new(self, x, success_expected: bool):
+        evaluation = baker.make(Evaluation, state=x, course__semester=self.semester)
+
+        semester_overview_page = self.app.get(f"/staff/semester/{self.semester.pk}", user=self.manager, status=200)
+
+        form = semester_overview_page.forms["evaluation_operation_form"]
+
+        form["evaluation"] = [evaluation.pk]
+
+        confirmation_page = form.submit(name="target_state", value=str(Evaluation.State.NEW))
+        confirmation_form = confirmation_page.forms["evaluation-operation-form"]
+        confirmation_form.submit()
+
+        self.assertTrue(success_expected, "Should have raised exception by now")
+        self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
```

Comment:
What do you think about the following?

```suggestion
        confirmation_form.submit(expect_errors=not success_expected)

        if success_expected:
            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
```

Reply:
This also leaves the question whether we want to do anything in the `else` case - I think it would also be fine like this

Reply:
(Agreed. `"Should have raised exception by now"` seems subtly wrong to me, exceptions wouldn't propagate into the test context here. I would have expected this to fail -- maybe the `assertRaises` below is even triggered by some unintended error)

Reply:
I changed it to:
```python
        confirmation_page = form.submit(name="target_state", value=str(Evaluation.State.NEW))

        if success_expected:
            confirmation_form = confirmation_page.forms["evaluation-operation-form"]
            confirmation_form.submit()
            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
        else:  # no confirmation form should exist
            self.assertNotIn("evaluation-operation-form", confirmation_page.forms)

```